python/hyperspace/hyperspace.py

```diff
@@ -0,0 +1,172 @@
+
+        >>> _getScalaSeqFromList(list)
+        """
+        java_import(self.jvm, "scala.collection.JavaConversions._")
+        java_import(self.jvm, "scala.collection.Seq")
+        java_import(self.jvm, 'java.util.*')
+        resultArrayList = self.jvm.ArrayList(len(list))
+        for element in list:
+            resultArrayList.add(self.jvm.String(element))
+        return self.jvm.scala.collection.JavaConverters.asScalaIteratorConverter(
+            resultArrayList.iterator()).asScala().toSeq()
+
+    def indexes(self):
+        """
+        Gets available indexes.
+        :return: dataFrame object containing list of indexes.
+
+        >>> hyperspace = Hyperspace(spark)
+        >>> hyperspace.indexes()
+        """
+        return DataFrame(self.hyperspace.indexes(), self.spark._wrapped)
+
+    def createIndex(self, dataFrame, indexConfig):
+        """
+        Creates index on the given dataframe using the given indexConfig.
+        :param dataFrame: dataFrame
+        :param indexConfig: indexConfig
+
+        >>> hyperspace = Hyperspace(spark)
+        >>> idxConfig = IndexConfig("indexName", ["c1"], ["c2","c3"])
+        >>> df = spark.read.parquet("./sample.parquet").toDF("c1", "c2", "c3")
+        >>> hyperspace.createIndex(df, indexConfig)
+        """
+        self.hyperspace.createIndex(dataFrame._jdf, self._getJavaIndexConfig(indexConfig))
+
+    def deleteIndex(self, indexName):
+        """
+        Soft deletes given index.
+        :param indexName: index name
+
+        >>> hyperspace = Hyperspace(spark)
+        >>> hyperspace.deleteIndex("indexname")
+        """
+        self.hyperspace.deleteIndex(indexName)
+
+    def restoreIndex(self, indexName):
+        """
+        Restores index with given index name.
+        :param indexName: index name
+
+        >>> hyperspace = Hyperspace(spark)
+        >>> hyperspace.restoreIndex("indexname")
+        """
+        self.hyperspace.restoreIndex(indexName)
+
+    def vacuumIndex(self, indexName):
+        """
+        Vacuums index with given index name.
+        :param indexName: index name
+
+        >>> hyperspace = Hyperspace(spark)
+        >>> hyperspace.vacuumIndex("indexname")
+        """
+        self.hyperspace.vacuumIndex(indexName)
+
+    def refreshIndex(self, indexName):
+        """
+        Update indexes for the latest version of the data.
+        :param indexName: index name
+
+        >>> hyperspace = Hyperspace(spark)
+        >>> hyperspace.refreshIndex("indexname")
+        """
+        self.hyperspace.refreshIndex(indexName)
+
+    def cancel(self, indexName):
+        """
+        Cancel api to bring back index from an inconsistent state to the last known stable state.
+        :param indexName: index name
+
+        >>> hyperspace = Hyperspace(spark)
+        >>> hyperspace.cancel("indexname")
+        """
+        self.hyperspace.cancel(indexName)
+
+    def explain(self, df, redirectFunc=lambda x: sys.stdout.write(x), verbose=False):
```

Comment:
Can you change the order of these params?  In .Net we're putting `verbose` before `redirectFunc`.

Reply:
Sure. Only concern is  this would be a breaking change with what is deployed in Synapse. In synapse python binding the signature is this. 
@imback82 @rapoth Just checking are you ok changing this signature here? Or should be a separate PR? It will take ~2 weeks to get the changed signature deployed in synapse.

Reply:
Yes, let's make it consistent with C# and Scala. Thanks!

Reply:
I'm fine with it. Let's make it consistent. Thanks!